src/components/Carousel/index.tsx

```diff
@@ -48,9 +63,20 @@ export default function Carousel({ items, onClose, onSelectItem, buttonText = "S
   const onSelect = useCallback(() => {
     const currentItem = items[currentItemIndex];
     if (onSelectItem && currentItem !== undefined) {
-      onSelectItem(currentItem.name);
+      const validatedSelection = isAbleToSelect(currentItem.name);
```

Comment:
Rather than let the user try to select the item and then displaying the toast message, it would be better to just disable the button if they can't purchase it.